src/fides/api/alembic/migrations/versions/31493e48c1d8_table_updates_for_user_invite.py

```diff
@@ -0,0 +1,70 @@
+"""table updates for user invite
+
+Revision ID: 31493e48c1d8
+Revises: 4fb779cc5f17
+Create Date: 2024-03-08 18:05:31.392727
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from citext import CIText
+
+# revision identifiers, used by Alembic.
+revision = "31493e48c1d8"
+down_revision = "4fb779cc5f17"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.execute("create type disabledreason as enum('pending_invite')")
+    op.create_table(
+        "fides_user_invite",
+        sa.Column("id", sa.String(length=255), nullable=False),
+        sa.Column(
+            "created_at",
+            sa.DateTime(timezone=True),
+            server_default=sa.text("now()"),
+            nullable=True,
+        ),
+        sa.Column(
+            "updated_at",
+            sa.DateTime(timezone=True),
+            server_default=sa.text("now()"),
+            nullable=True,
+        ),
+        sa.Column("username", CIText(), nullable=False),
+        sa.Column("hashed_invite_code", sa.String(), nullable=False),
+        sa.Column("salt", sa.String(), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["username"], ["fidesuser.username"], ondelete="CASCADE"
+        ),
+        sa.PrimaryKeyConstraint("id", "username"),
+    )
+    op.create_index(
+        op.f("ix_fides_user_invite_id"), "fides_user_invite", ["id"], unique=False
+    )
+    op.add_column("fidesuser", sa.Column("email_address", CIText(), nullable=True))
+    op.add_column("fidesuser", sa.Column("disabled", sa.Boolean(), nullable=False))
+    op.add_column(
+        "fidesuser",
+        sa.Column(
+            "disabled_reason",
+            sa.Enum("pending_invite", name="disabledreason"),
+            nullable=True,
+        ),
+    )
+    op.create_unique_constraint(
+        "fidesuser_email_address", "fidesuser", ["email_address"]
```

Comment:
does this play well with null values? i don't _think_ nulls are treated as distinct, are they? this is important, especially since this is a new column and existing records won't have entries

in general, i'd be sure to test this migration pretty well on your local db populated with `fidesuser`s, feels like a few  potential issues lurking here, unless i'm missing something (haven't tested myself!)

Reply:
That's correct, the unique constraint only applies to non-null values, so the existing rows should be fine with null values for `email_address`

Reply:
Ok, here's the local testing I did

1. Checked out `main` and created two users in the database
2. Checked out this branch and ran the migration
3. Migration was successful, neither of the existing users have an `email_address` and both have the default value of False for the `disabled` column

Reply:
ok great! thanks for adjusting and testing that 👍